app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/settings/SettingsDownloadsLimitScreen.kt

```diff
@@ -51,13 +58,17 @@ fun SettingsDownloadsLimitScreen(navigateBack: (() -> Unit)?) {
         ) {
             SettingTitle(titleRes = R.string.settingsDownloadsLimitTitle)
 
-            val (selectedItem, setSelectedItem) = rememberSaveable { mutableIntStateOf(0) } // TODO: Use DataStore or Realm
-            SingleSelectOptions(DownloadsLimit.entries, { selectedItem }, setSelectedItem)
+            var selectedItem by rememberSaveable { mutableIntStateOf(downloadLimit.ordinal) }
+            SingleSelectOptions(DownloadsLimitOption.entries, { selectedItem }, { position ->
+                selectedItem = position
+                val selectedDownloadLimit = DownloadLimit.entries[position]
+                onDownloadLimitChange(selectedDownloadLimit)
+            })
         }
     }
 }
 
-enum class DownloadsLimit(
+enum class DownloadsLimitOption(
```

Comment:
Do we prefer `DownloadLimit` or `DownloadsLimit` ? _(with or without the -s)_
Because we are using `DownloadLimit` line 46 and `DownloadsLimitOption` line 71.